modules/ROOT/pages/dataweave-streaming.adoc

```diff
@@ -0,0 +1,398 @@
+     name: payload.name
+ }
+----
+
+The script fails with the following error:
+
+[source,error,linenums]
+----
+4| input payload application/json streaming=true
+         ^^^^^^^
+Parameter `payload` is not stream capable.
+Reasons:
+ - Variable payload is referenced more than once. Locations:
+---------------------------
+
+8|      family: payload.family filter (member) -> member.age > 3,
+                ^^^^^^^
+---------------------------
+
+9|      name: payload.name
+              ^^^^^^^ at
+4| input payload application/json streaming=true
+----
+
+
+==== Error: Wrong Scope Reference
+
+Validation fails if a script attempts to reference a variable from a scope that is different from the scope in which the variable is defined.
+
+The following script fails because the `payload` variable is referenced from within the lambda expression `[1,2,3] map ((item, index) -> payload)`. Even if the expression is `[1] map ((item, index) -> payload`, streaming fails because `payload` is in the wrong scope.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+
+@StreamCapable()
+input payload application/json
+output application/json
+---
+[1,2,3] map ((item, index) -> payload)
+---
+
+The example fails with the following error:
+
+[source,error,linenums]
+----
+4| input payload application/json
+         ^^^^^^^
+Parameter `payload` is not stream capable.
+Reasons:
+ - Variable payload is referenced in a different scope from where it was defined. Locations:
+---------------------------
+
+9| [1,2,3] map ((item, index) -> payload)
+                ^^^^^^^^^^^^^^^^^^^^^^^^ at
+4| input payload application/json
+----
+
+[[stream_output]]
+== Streaming Output
+
+After processing streamed data, you can stream the output directly to another message processor. To facilitate this behavior, use the `deferred` writer property in the output directive of the DataWeave script, for example, `output application/json deferred=true`.
+
+Building on the example in <<stream_json, JSON streaming>>, the following flow uses a DataWeave script to filter streamed input and then streams the output directly to a Write operation.
+
+[source,DataWeave,linenums]
+----
+<flow name="dw-streamingexample">
+  <file:listener doc:name="On New or Updated File"
+    config-ref="File_Config" directory="/Users/me/testing/json" recursive="false" outputMimeType="application/json;
+    streaming=true">
+    <scheduling-strategy>
+      <fixed-frequency timeUnit="SECONDS" />
+    </scheduling-strategy>
+    <file:matcher />
+  </file:listener>
+  <ee:transform doc:name="Transform Message">
+    <ee:message>
+      <ee:set-payload><![CDATA[%dw 2.0
+
+@StreamCapable()
+input payload application/json
+output application/json deferred = true
+---
+{
+   family: payload.family filter (member) -> member.age > 1
+}]]></ee:set-payload>
+    </ee:message>
+  </ee:transform>
+  <file:write doc:name="Write"
+      config-ref="File_Config2"
+      path="/Users/me/testing/output.json"/>
+</flow>
+----
+
```

Comment:
Provide a transitional sentence or frag. between the example and this ordered list?

Reply:
done